source/_posts/2017-03-30-Haptic-feedback-makes-you-vibrate.md

```diff
@@ -0,0 +1,147 @@
+heavyImpactFeedbackGenerator.impactOccurred()
+
+```
+
+### Notification
+
+The notification type of feedback should be used to indicate to the user that a task has completed. Depending on the outcome of the task (success, error, warning), the feedback pattern differs.
+
+**Success**
+
+<video width="400" controls>
+  <source src="http://nodesmanager.io/haptic/success.mp4" type="video/mp4">
+  Your browser does not support HTML5 video.
+</video>
+
+```swift
+let successNotificationFeedbackGenerator = UINotificationFeedbackGenerator()
+successNotificationFeedbackGenerator.notificationOccurred(.success)
+```
+
+**Warning**
+
+<video width="400" controls>
+  <source src="http://nodesmanager.io/haptic/warning.mp4" type="video/mp4">
+  Your browser does not support HTML5 video.
+</video>
+
+```swift
+let warningNotificationFeedbackGenerator = UINotificationFeedbackGenerator()
+warningNotificationFeedbackGenerator.notificationOccurred(.warning)
+```
+
+**Failure**
+
+<video width="400" controls>
+  <source src="http://nodesmanager.io/haptic/error.mp4" type="video/mp4">
+  Your browser does not support HTML5 video.
+</video>
+
+```swift
+let errorNotificationFeedbackGenerator = UINotificationFeedbackGenerator()
+errorNotificationFeedbackGenerator.notificationOccurred(.error)
+```
+
+
+Using the `UIFeedbackGenerator` may have a little latency. If you want to use it to match some sound or visual effects, you can call the `prepare()` method on the generator to put the Taptic Engine in a prepared state and reduce the latency. It will stau prepared a few seconds, or until the next feedback is triggered.
+
+
+## When to use haptic feedback
+
+We saw that creating haptic feedback is super easy. However, the complicated part is to know when and where to use it.
+
+As usual, Apple helps with this, by adding a section on it to the [Human Interface Guidelines](https://developer.apple.com/ios/human-interface-guidelines/interaction/feedback/). And we can also learn a lot by looking at where the haptic feedback is used across iOS. 
+
+We all know by now that the Home button isn't an actual button anymore, it's just some very smartly done and well-placed haptic feedback. But that specific patter doesn't come by default with the new `UIFeedbackGenerator`. Other parts of iOS though use the exact types of feedback that the developers can access through the new APIs. 
+
+For example, when you set an alarm or a timer, you'll notice a light tap whenever the selected time changes. Or when you type, and you hold down a letter on the keyboard to get to the various accents, the same light tap will inform you when you selected a different character.
+
+Some controls, such as the `UIRefreshControl` have the feedback build in, and you will feel a small "knock" when you pull to refresh a scroll view. Same for zooming in or out a `UIScrollView`, zooming too much (either in or out) will also create a "knock".
```

Comment:
The comma after UIScrollView should be something else but I'm not sure exactly what, either ; : or - 

I think it's - but I could be wrong